code/datums/ai/babies/babies_subtrees.dm

```diff
@@ -18,7 +18,7 @@
 		controller.queue_behavior(reproduce_behavior, BB_BABIES_TARGET, BB_BABIES_CHILD_TYPES)
 		return SUBTREE_RETURN_FINISH_PLANNING
 
-	if(controller.pawn.gender == FEMALE || !controller.blackboard[BB_BREED_READY])
+	if(QDELETED(controller.pawn) || controller.pawn.gender == FEMALE || !controller.blackboard[BB_BREED_READY])
```

Comment:
It seems very silly to add this individually to a bunch of subtrees as opposed to just moving this up one layer to "before we call `SelectBehaviours` on any subtree"
Subtrees _should_ just be queueing rather than executing actions if I remember the flow correctly, so if you're deleted at any step here you'll be deleted in all of them and it doesn't make sense to do any planning.

Reply:
Sorry for not replying to this I forgot
but yeah what you have done in these later commits looks like the right approach to me, there's no reason AI behaviours should continue planning or running when the mob is deleted